draft of release: 2025-10-05-1.3.71.md (#137)
fiskaltrust/release-notes: middleware/2025-10-05-1.3.71.md

---
authors: poscreator
slug: middleware/1.3.71
tags: [Middleware, Germany, Italy]
---

> # Middleware 1.3.71 (Germany, Austria, Europe, Spain, Portugal, Greece)

Reviewer: ```suggestion
# Middleware 1.3.71 (Germany, Austria, European Union, Spain, Portugal, Greece)
```

In this version of the Middleware, our focus was on implementing Queue EU and also resolving some  bugs from the queues and SCUs in Germany and Austria.
<!--truncate-->

:::caution

Version 1.3 of the Middleware is meant for the German and Italian markets only, customers in Austria and France should continue to use version 1.2.
We are currently testing a unified version for all markets, and will release an update about this soon.

:::
## Feature: Introduce QueueEU
??

> ## Bug Fix: Serialization with System.Text.Json contains escaped characters

Reviewer: ```suggestion
## 🇪🇸🇵🇹🇬🇷 Bug Fix: Serialization with System.Text.Json contains escaped characters
```

> Disable character escaping for QueueItem (both request and response), and wherever necessary to ensure compatibility with external systems.

Reviewer: ```suggestion
Fix character escaping for QueueItem (both request and response), and wherever necessary to ensure compatibility with external systems.
```

---
authors: poscreator
slug: middleware/1.3.71
tags: [Middleware, Germany, Italy]
---

# Middleware 1.3.71 (Germany, Austria, Europe, Spain, Portugal, Greece)
In this version of the Middleware, our focus was on implementing Queue EU and solving some minor issues in queues and SCUs in Germany and Austria.
<!--truncate-->

:::caution

> Version 1.3 of the Middleware is meant for the German and Italian markets only, customers in Austria and France should continue to use version 1.2.
> We are currently testing a unified version for all markets, and will release an update about this soon.

Reviewer: ```suggestion
Version 1.3 of the Middleware is meant for the German and Italian markets only (with Spain, Grece and Portugal currently in the works).
Customers in Austria and France should continue to use version 1.2.
We are currently testing a unified version for all markets.
```

:::
## Feature: Introduce QueueEU
??
## Bug Fix: Serialization with System.Text.Json contains escaped characters

Disable character escaping for QueueItem (both request and response), and wherever necessary to ensure compatibility with external systems.

### Affected packages:

> - _fiskaltrust.Middleware.Queue.*

Reviewer: ```suggestion
- _fiskaltrust.Middleware.Queue.*_
```

  
## 🇩🇪 Bug Fix: JournalFunction provided incorrect details

The journal endpoint /json/v0/Journal?type=0 is intended to return detailed information about the middleware version.

:::info

Previously, this endpoint returned a generic name and version that did not accurately reflect the middleware in use. This issue has now been resolved, and the endpoint correctly returns the specific middleware information.

:::

### Affected packages:

> - _fiskaltrust.Middleware.Queue.*

Reviewer: ```suggestion
- _fiskaltrust.Middleware.Queue.*_
```

> ## 🇩🇪 Bug Fix: Queue SQLite always showed a pending database on startup

Reviewer: ```suggestion
## 🌐 Bug Fix: Queue SQLite always showed a pending database update on startup
```

> The SQLite queue previously indicated a pending database update every time it started, even after the update had already been applied. This has now been fixed—the correct version is now saved in the database, preventing repeated updates on subsequent startups.

Reviewer: ```suggestion
The SQLite queue previously indicated a pending database update every time it started, even after the update had already been applied. This has now been fixed. The correct version is now saved in the database, preventing repeated updates on subsequent startups.
```

> :::

Reviewer: ```suggestion
```

### Affected packages:

> - _fiskaltrust.Middleware.Queue.SQLite

Reviewer: ```suggestion
- _fiskaltrust.Middleware.Queue.SQLite_
```

## 🇩🇪 Bug Fix: SwissbitCloud throwing errors on startup

> The SwissbitCloud SCU was throwing startup errors that didn’t impact its actual functionality, like "Failed to execute GetTseInfoAsync"

Reviewer: ```suggestion
The SwissbitCloud SCU was throwing startup errors that didn’t impact its actual functionality, like `Failed to execute GetTseInfoAsync`
```

:::info

These errors occurred because the system attempted to request the TseInfo before the FCC had fully initialized. Since FCC initialization happens only once per system, the process has now been updated to wait until the FCC setup is complete before making the request.
:::

### Affected packages:

> - _fiskaltrust.Middleware.SCU.DE.SwissbitCloud

Reviewer: ```suggestion
- _fiskaltrust.Middleware.SCU.DE.SwissbitCloud_
```

## 🇩🇪 Bug Fix: Fix logtimeformat casing in DE.Epson SCU

> The issue is that the Epson SCU uses "UnixTime" (with a capital "U" and "T") in the QR code payload, whereas it should be "unixtime" in lowercase—this mismatch causes problems with the FinanzCheck app.

Reviewer: ```suggestion
We've fixed an issue where previously on DE.Epson TSEs the Log Time Format was set to "UnixTime" instead of "unixtime".
```

> :::

Reviewer: ```suggestion
```

### Affected packages:

> - _fiskaltrust.Middleware.SCU.DE.Epson

Reviewer: ```suggestion
- _fiskaltrust.Middleware.SCU.DE.Epson_
```

> ## AT Improvement: Use v0 of IATSSCD 

Reviewer: ```suggestion
## 🇦🇹 Bug Fix: Fix OnlineSCU access from the Queue
```

> The SCUs version 1.3 supports the online SCU endpoint:  signing[-sandbox].fiskaltrust.at.

Reviewer: ```suggestion
We've fixed a problem the austrian 1.3 implementation had when accessing the OnlineSCU.
```

:::info

> The online SCU: signing[-sandbox].fiskaltrust.at is still running on 1.2.

Reviewer: ```suggestion
The 1.3 version of the middleware is in preview for the austrian market.
:::
```

### Affected packages:

> - _fiskaltrust.Middleware.Queue.*

Reviewer: ```suggestion
- _fiskaltrust.Middleware.Queue.*_
```

> ## DE Improvement: End2End Testing for SwissbitCloud V2
> 
> SCUConfiguration related to SwissbitCloudV2 has been updated to contain required endpoints and certificationId.
> 
> ### Affected packages:
> - _fiskaltrust.Middleware.SCU.DE.SwissbitCloudV2

Reviewer: ```suggestion
## 🇩🇪 New SCU: SwissbitCloud V2 SCU is now available

The new SwissbitCloudV2 SCU is now available in the german market and can be rolled out.

Since this SCU is fully hosted in the cloud it's not using a local component like the FCC that the old SwissbitCloud had making operation of this new version a lot simpler.

### Affected packages:
- _fiskaltrust.Middleware.SCU.DE.SwissbitCloudV2_
```

> ## DE Bug Fix: Nullpointer on timerange export

Reviewer: ```suggestion
## 🇩🇪 Bug Fix: Nullpointer on timerange export
```

> For a very rare case we needed to increase the search span for the actionjournal.

Reviewer: ```suggestion
We've fixed a rare nullpointer exception during a DSFinV-K export that occurred on some setups.
```

### Affected packages:

> - _fiskaltrust.Middleware.Queue.*

Reviewer: ```suggestion
- _fiskaltrust.Middleware.Queue.*_
```

> ## End2End Testing for SwissbitCloud V2
> 
> We have finished final tests for SwissbitCloud V2 and updated settings for production.
> 
> ### Affected packages:
> - _fiskaltrust.Middleware.SCU.DE.SwissbitCloudV2

Reviewer: ```suggestion

> As always, we're happy to receive feedback and suggestions directly via issues in our [GitHub repositories](https://github.com/fiskaltrust).
```
